src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1438,6 +1438,17 @@ void LinearScan::makeUpperVectorInterval(unsigned varIndex)
 {
     Interval* lclVarInterval = getIntervalForLocalVar(varIndex);
     assert(Compiler::varTypeNeedsPartialCalleeSave(lclVarInterval->registerType));
+    if (lclVarInterval->registerType == TYP_SIMD64)
+    {
+        Interval* newInt        = newInterval(TYP_SIMD64);
+        newInt->relatedInterval = lclVarInterval;
+        newInt->isUpperVector   = true;
+        setIntervalAsSpilled(newInt);
```

Comment:
You need to fix this in `allocateRegisters()`. See my comments above.

Reply:
Thanks for the pointers. I modified in `allocateRegisters()` as you suggested and it works as expected. 


```
else if (lclVarInterval->registerType == TYP_SIMD64)
{
    allocate = false;
     lclVarInterval->isSpilled = true;
 }
```


Re your comment about `IsSpilled` , I assume you did not mean adding a new flag to keep tract of SIMD64 specific spills. We already have an `IsSpilled` which seem to track  spilled to mem cases. I set that


I run into some other issues with debug build due to this  but I'm looking at it. The issue is, we still assign a register and that seem to cause issues later
https://github.com/dotnet/runtime/blob/53bb518ea482a5960577b151c11cc1297e5dea6d/src/coreclr/jit/lsra.cpp#L4830-L4831

Reply:
Rethinking, perhaps we should just reuse the `isPartiallySpilled` flag. From ABI perspective, that is what we are trying to do (spill partial zmm register) except for the optimization sake, we decide to spill entire contents instead of "partial". It will be much simpler with that and you will set `allocate = false` regardless of if it was `isPartiallySpilled` or not for `TYP_SIMD64`.

Reply:
I updated the changes now - https://github.com/dotnet/runtime/pull/80960/commits/5e7b2e9e74cc8828261b69838230301916a48fe3 . The tests ran fine locally for me but till wait  everything hopefully passes before asking for a  new review